### PR TITLE
Add Python 3.7 to travis.yml

### PR DESCRIPTION
**Pull request by mrocklin**
- [ ] Tests added / passed
- [ ] Passes `flake8 dask`

### CODE REVIEW

**Comment by mrocklin** [comment]
@TomAugspurger what is the right behavior for this:

```
  /home/travis/build/dask/dask/dask/dataframe/methods.py:333: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version
  of pandas will change to not sort by default.
  
  To accept the future behavior, pass 'sort=False'.
  
  To retain the current behavior and silence the warning, pass 'sort=True'.
  
    out = pd.concat(dfs3, join=join)
```

File: .travis.yml [hunk 1]
```diff
@@ -15,15 +15,6 @@ _base_envs:
...
   fast_finish: true
   include:
-    - env:
-      - PYTHON=2.7
-      - NUMPY=1.14.1
-      - PANDAS=0.22.0
-      - *test_and_lint
-      - *no_coverage
-      - *no_optimize
-      - *no_imports
```
Comment (mrocklin) [inline .travis.yml]: Does anyone mind me dropping this CI entry as we add one for 3.7?  cc @jakirkham @jcrist

Comment (jakirkham) [reply .travis.yml]: I do. We generally try to support both Python 2/3 in our codebase as there are inevitably scientific libraries that we want to pull in that haven't made the switch. Though I think this has gotten better in the past year or two, it's still a surprisingly common problem (even with new codes).

Comment (mrocklin) [reply .travis.yml]: There is still another 2.7 entry.  We used to have two.  You might want to check the full file in this branch

Comment (mrocklin) [reply .travis.yml]: @jakirkham does your concern still hold, given that there is another Python 2.7 entry?

Comment (jakirkham) [reply .travis.yml]: Oops. Sorry. Missed the other one. This is fine.

**Comment by mrocklin** [comment]
OK, merging this in.

**Pull request merged**